src/longRange/Planar.cpp

```diff
@@ -184,7 +209,30 @@ void Planar::readXML(XMLfileUnits& xmlconfig)
 	this->init();
 }
 
-void Planar::calculateLongRange(){
+void Planar::calculateLongRange() {
+
+	if (_region.refPosID[0]+_region.refPosID[1] > 0) {
+		if (not _DCregistered) {
+			_subject = getSubject();
+			this->update(_subject);
+			if (nullptr != _subject) {
+				_subject->registerObserver(this);
+#ifndef NDEBUG
+				global_log->info() << "Long Range Correction: Subject registered" << endl;
+#endif
+			} else {
+				global_log->error() << "Long Range Correction: Initialization of plugin DistControl is needed before! Program exit..." << endl;
+				Simulation::exit(-1);
+			}
+			_DCregistered = true;
+		} else {
+			this->update(_subject);
+#ifndef NDEBUG
+			global_log->info() << "Long Range Correction: Subject already registered" << endl;
+#endif
+		}
+	}
+
```

Comment:
The registration of the Observer to the Subject should be done in readXML(). Once the Observer is registered, the update() method is automatically called by the Subject, when it changes its state (i.e. in this case when DistControl (the Subject) calculates new interface positions).
You dont need to call update() explicitly, this is done by method Subject::informObservers().